include/deal.II/base/exceptions.h

```diff
@@ -1130,6 +1130,25 @@ namespace StandardExceptions
 #define AssertIsFinite(number) Assert(dealii::numbers::is_finite(number), \
                                       dealii::ExcNumberNotFinite(std::complex<double>(number)))
 
+#ifdef DEAL_II_WITH_CUDA
+/**
+ * An assertion that checks that the error code produced by calling a CUDA
+ * routine is equal to cudaSuccess.
+ *
+ * @ingroup Exceptions
+ * @author Bruno Turcksin, 2016
+ */
+#define CudaAssert(error_code)                                                \
+  {                                                                           \
+    if (cudaSuccess != error_code)                                            \
+      {                                                                         \
+        fprintf(stderr,"Error in %s (%d): %s\n",__FILE__,                       \
+                __LINE__,cudaGetErrorString(error_code));                       \
+        exit(1);                                                                \
```

Comment:
I think that the problem is that ```DEBUG``` is not set properly when we use nvcc. So the assert does nothing. When I use the ```Assert``` in ```templates.h```, I guess gcc expands the macro and it works.